cannot be cast to class io.appium.java_client.android.AndroidDriver while using EventFiringDecorator

## Description

I'm trying to use the decorator mechanism provided by selenium 4.
But following the doc to decorate the android driver, I get the exception

java.lang.ClassCastException: class jdk.proxy2.$Proxy22 cannot be cast to class io.appium.java_client.android.AndroidDriver (jdk.proxy2.$Proxy22 is in module jdk.proxy2 of loader 'app'; io.appium.java_client.android.AndroidDriver is in unnamed module of loader 'app')


## Environment

Java client build version or git revision if you use some snapshot: 8.0.0
Appium server version or git revision if you use some snapshot: 1.22.3
Desktop OS/version used to run Appium if necessary: Mac Monterey
Node.js version (unless using Appium.app|exe) or Appium CLI or Appium.app|exe: Appium.exe
Mobile platform/version under test: android
Real device or emulator/simulator: Both

## Details

Please provide more details, if necessary.


## Code To Reproduce Issue [ Good To Have ]

 AndroidDriver driver = new AndroidDriver(url, driverCapabilities);
 driver=(AndroidDriver) new EventFiringDecorator(new MyLIstner()).decorate(driver);


SImilar to Cannot decorate AndroidDriver (probably the same for IOSDriver #1643

Also i meet this issue in Appium:java-client:8.1.1，the following is callding code:

WebDriverListener webDriverListener = new CustomListener();
appiumDriver = (AppiumDriver) new EventFiringDecorator(webDriverListener).decorate(appiumDriver);

Caused by: java.lang.ClassCastException: class net.bytebuddy.renamed.java.lang.Object$ByteBuddy$AuHk1O9o cannot be cast to class io.appium.java_client.AppiumDriver (net.bytebuddy.renamed.java.lang.Object$ByteBuddy$AuHk1O9o is in unnamed module of loader net.bytebuddy.dynamic.loading.ByteArrayClassLoader @6af310c7; io.appium.java_client.AppiumDriver is in unnamed module of loader 'app')

--- a/gradle.properties
+++ b/gradle.properties
@@ -7,4 +7,4 @@ signing.secretKeyRingFile=PathToYourKeyRingFile
 ossrhUsername=your-jira-id
 ossrhPassword=your-jira-password
 
-selenium.version=4.2.1
+selenium.version=4.4.0

--- a/src/main/java/io/appium/java_client/remote/AppiumW3CHttpCommandCodec.java
+++ b/src/main/java/io/appium/java_client/remote/AppiumW3CHttpCommandCodec.java
@@ -22,22 +22,13 @@ import static org.openqa.selenium.remote.DriverCommand.GET_ELEMENT_LOCATION_ONCE
 import static org.openqa.selenium.remote.DriverCommand.GET_ELEMENT_SIZE;
 import static org.openqa.selenium.remote.DriverCommand.GET_PAGE_SOURCE;
 import static org.openqa.selenium.remote.DriverCommand.IS_ELEMENT_DISPLAYED;
-import static org.openqa.selenium.remote.DriverCommand.SEND_KEYS_TO_ACTIVE_ELEMENT;
 import static org.openqa.selenium.remote.DriverCommand.SEND_KEYS_TO_ELEMENT;
 import static org.openqa.selenium.remote.DriverCommand.SET_TIMEOUT;
 import static org.openqa.selenium.remote.DriverCommand.SUBMIT_ELEMENT;
 
-import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableMap;
-
-import org.openqa.selenium.interactions.KeyInput;
-import org.openqa.selenium.interactions.Sequence;
 import org.openqa.selenium.remote.codec.w3c.W3CHttpCommandCodec;
 
-import java.util.Collection;
 import java.util.Map;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 public class AppiumW3CHttpCommandCodec extends W3CHttpCommandCodec {
     /**
@@ -51,7 +42,6 @@ public class AppiumW3CHttpCommandCodec extends W3CHttpCommandCodec {
         defineCommand(GET_ELEMENT_ATTRIBUTE, get("/session/:sessionId/element/:id/attribute/:name"));
         defineCommand(IS_ELEMENT_DISPLAYED, get("/session/:sessionId/element/:id/displayed"));
         defineCommand(GET_PAGE_SOURCE, get("/session/:sessionId/source"));
-        defineCommand(SEND_KEYS_TO_ACTIVE_ELEMENT, post("/session/:sessionId/actions"));
     }
 
     @Override
@@ -76,23 +66,6 @@ public class AppiumW3CHttpCommandCodec extends W3CHttpCommandCodec {
     protected Map<String, ?> amendParameters(String name, Map<String, ?> parameters) {
         // This blocks parent constructor from undesirable parameters amending
         switch (name) {
-            case SEND_KEYS_TO_ACTIVE_ELEMENT:
-                Object rawValue = parameters.get("value");
-                //noinspection unchecked
-                Stream<CharSequence> source = (rawValue instanceof Collection)
-                        ? ((Collection<CharSequence>) rawValue).stream()
-                        : Stream.of((CharSequence[]) rawValue);
-                String text = source.collect(Collectors.joining());
-
-                final KeyInput keyboard = new KeyInput("keyboard");
-                Sequence sequence = new Sequence(keyboard, 0);
-                for (int i = 0; i < text.length(); ++i) {
-                    sequence.addAction(keyboard.createKeyDown(text.charAt(i)))
-                            .addAction(keyboard.createKeyUp(text.charAt(i)));
-                }
-                return ImmutableMap.<String, Object>builder()
-                        .put("actions", ImmutableList.of(sequence.toJson()))
-                        .build();
             case SEND_KEYS_TO_ELEMENT:
             case SET_TIMEOUT:
                 return super.amendParameters(name, parameters);

